pkg/clusterapi/aws/actuator.go

```diff
@@ -0,0 +1,403 @@
+		&ec2.BlockDeviceMapping{
+			DeviceName: aws.String("/dev/sda1"),
+			Ebs: &ec2.EbsBlockDevice{
+				DeleteOnTermination: aws.Bool(true),
+				VolumeSize:          aws.Int64(100),
+				VolumeType:          aws.String("gp2"),
+			},
+		},
+		&ec2.BlockDeviceMapping{
+			DeviceName: aws.String("/dev/sdb"),
+			Ebs: &ec2.EbsBlockDevice{
+				DeleteOnTermination: aws.Bool(true),
+				VolumeSize:          aws.Int64(100),
+				VolumeType:          aws.String("gp2"),
+			},
+		},
+	}
+
+	bootstrapKubeConfig, err := getBootstrapKubeconfig()
+	if err != nil {
+		return nil, err
+	}
+	userData := getUserData(bootstrapKubeConfig, coMachineSet.Spec.Infra)
+	userDataEnc := base64.StdEncoding.EncodeToString([]byte(userData))
+
+	inputConfig := ec2.RunInstancesInput{
+		ImageId:      describeAMIResult.Images[0].ImageId,
+		InstanceType: aws.String(coMachineSet.Spec.Hardware.AWS.InstanceType),
+		MinCount:     aws.Int64(1),
+		MaxCount:     aws.Int64(1),
+		UserData:     &userDataEnc,
+		KeyName:      aws.String(coCluster.Spec.Hardware.AWS.KeyPairName),
+		IamInstanceProfile: &ec2.IamInstanceProfileSpecification{
+			Name: aws.String(defaultIAMRole),
+		},
+		BlockDeviceMappings: blkDeviceMappings,
+		TagSpecifications:   []*ec2.TagSpecification{tagInstance},
+		NetworkInterfaces:   networkInterfaces,
+	}
+
+	runResult, err := svc.RunInstances(&inputConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	return runResult, nil
+}
+
+// Delete method is used to delete a AWS machine
+func (a *AWSActuator) Delete(machine *clusterv1.Machine) error {
+	/*
+		var (
+			err       error
+			machineID = d.decodeMachineID(d.MachineID)
+		)
+
+		svc := d.createSVC()
+		input := &ec2.TerminateInstancesInput{
+			InstanceIds: []*string{
+				aws.String(machineID),
+			},
+			DryRun: aws.Bool(true),
+		}
+		_, err = svc.TerminateInstances(input)
+		awsErr, ok := err.(awserr.Error)
+		if ok && awsErr.Code() == "DryRunOperation" {
+			input.DryRun = aws.Bool(false)
+			output, err := svc.TerminateInstances(input)
+			if err != nil {
+				glog.Errorf("Could not terminate machine: %s", err.Error())
+				return err
+			}
+
+			vmState := output.TerminatingInstances[0]
+			//glog.Info(vmState.PreviousState, vmState.CurrentState)
+
+			if *vmState.CurrentState.Name == "shutting-down" ||
+				*vmState.CurrentState.Name == "terminated" {
+				return nil
+			}
+
+			err = errors.New("Machine already terminated")
+		}
+
+		glog.Errorf("Could not terminate machine: %s", err.Error())
+		return err
+	*/
+	return nil
+}
+
+// Update the machine to the provided definition.
+// TODO: For now, update is a No-op
+func (a *AWSActuator) Update(c *clusterv1.Cluster, machine *clusterv1.Machine) error {
+	return nil
+}
+
+// Checks if the machine currently exists.
+func (a *AWSActuator) Exists(*clusterv1.Machine) (bool, error) {
+	return false, nil
+}
+
+// Helper function to create SVC
+func (a *AWSActuator) createSVC(region string) (*ec2.EC2, error) {
```

Comment:
SVC here made me think kube service, is there a better term we could use for this?

Reply:
Will fix